core/src/main/kotlin/fr/sncf/osrd/utils/CachedBlockMRSPBuilder.kt

```diff
@@ -0,0 +1,44 @@
+package fr.sncf.osrd.utils
+
+import fr.sncf.osrd.api.pathfinding.makePathProps
+import fr.sncf.osrd.envelope.Envelope
+import fr.sncf.osrd.envelope_sim.PhysicsRollingStock
+import fr.sncf.osrd.envelope_sim_infra.MRSP
+import fr.sncf.osrd.sim_infra.api.Block
+import fr.sncf.osrd.sim_infra.api.BlockId
+import fr.sncf.osrd.sim_infra.api.BlockInfra
+import fr.sncf.osrd.sim_infra.api.RawInfra
+import fr.sncf.osrd.utils.units.Offset
+import fr.sncf.osrd.utils.units.meters
+
+/** Used to compute block MRSPs and min time required to reach a point, with proper caching */
+data class CachedBlockMRSPBuilder(
+    val rawInfra: RawInfra,
+    val blockInfra: BlockInfra,
+    val rollingStock: PhysicsRollingStock?,
+) {
+    private val mrspCache = mutableMapOf<BlockId, Envelope>()
+
+    // 320km/h as default value (global max speed in France)
+    private val rsMaxSpeed = rollingStock?.maxSpeed ?: (320.0 / 3.6)
```

Comment:
Comment disappeared for some reason.
This could be a pb in the future for other countries. For example, germany's top train speed is 550hkm/h, which is a lot higher than France's. Not saying we could /should handle this right now, but we'll have to keep it in mind.

Reply:
I don't think German's train lines are equipped for trains to go faster than 300km/h.

Ideally we'd change the editoast -> core API to include some simulation data, but that seems a little overkill here. The whole process does a lot of approximations anyway.